cluster-autoscaler/cloudprovider/hetzner/hetzner_cloud_provider.go

```diff
@@ -206,6 +208,63 @@ func BuildHetzner(_ config.AutoscalingOptions, do cloudprovider.NodeGroupDiscove
 			klog.Fatalf("Failed to get servers for for node pool %s error: %v", nodegroupSpec, err)
 		}
 
+		// Get this node's cloud-init
+		var cloudInitBase64 string
+		nodeCloudInitVar := fmt.Sprintf("HCLOUD_%s_CLOUD_INIT", strings.ToUpper(spec.name))
```

Comment:
I did wonder about taking this semi-magical approach to be honest. Upper-casing was really for aesthetic purposes to avoid HCLOUD_pool_CLOUD_INIT but that's not that important. I actually think a better approach probably is to add some additional fields to the `--nodes` parameter as it removes the need to things like this.